### PR TITLE
MML-132 - Nested forms should not be allowed

### PR DESCRIPTION
**Pull request by sribeiro-daitan**

### CODE REVIEW

File: src/main/java/org/symphonyoss/symphony/messageml/elements/Element.java [hunk 1]
```diff
@@ -578,4 +588,8 @@ private String getElementsNameByClassName(Collection<Class<? extends Element>> e
   private String getElementNameByClass(Class<? extends Element> element) {
     return element.equals(TextNode.class) ? "text content" : element.getSimpleName().toLowerCase();
   }
+
+  private boolean hasElementAsChild(Collection<Class<? extends Element>> elementTypes) {
```
Comment (psanchez-daitangroup) [inline src/main/java/org/symphonyoss/symphony/messageml/elements/Element.java]: please add a java doc here

File: src/test/java/org/symphonyoss/symphony/messageml/elements/form/FormTest.java [hunk 2]
```diff
@@ -54,7 +54,16 @@ public void testFormWithInvalidChild() throws Exception {
     expectedException.expectMessage("Invalid MessageML content at element \"invalid-child\"");
 
     context.parseMessageML(input, null, MessageML.MESSAGEML_VERSION);
-    
+  }
+
+  @Test
```
Comment (cristiadu) [inline src/test/java/org/symphonyoss/symphony/messageml/elements/form/FormTest.java]: Please include another test with a form element inside a div inside another form element so we know the parentValidation is working as expected for other elements within the tree.

Comment (sribeiro-daitan) [reply src/test/java/org/symphonyoss/symphony/messageml/elements/form/FormTest.java]: @cristiadu the `testNestedForms()` method below does it. :)

Comment (cristiadu) [reply src/test/java/org/symphonyoss/symphony/messageml/elements/form/FormTest.java]: Weird, it was showing the old code to me for some reason.

**Pull request merged**